c_src/unifex/cnode/unifex/unifex.c

```diff
@@ -1,9 +1,22 @@
 #include "unifex.h"
 #include "cnode.h"
 
-void *unifex_alloc(size_t size) { return malloc(size); }
+UnifexEnv *unifex_alloc_env() {
+  UnifexEnv *env = (UnifexEnv *)unifex_alloc(sizeof(UnifexEnv));
+  *env = (UnifexEnv){.node_name = calloc(sizeof(char), 256),
+                     .ei_socket_fd = ERL_ERROR,
+                     .listen_fd = -1,
+                     .reply_to = NULL,
+                     .state = NULL,
+                     .released_states = NULL,
+                     .error = NULL};
+  return env;
+}
```

Comment:
I am not sure if this makes any sense.
In fact, after creating env with `unifex_alloc_env` it doesn't contain any useful data and cannot be used for e.g. sending messages from C node to Elixir process. 

I think that we are not able to avoid conditional compilation in this case so the user code that supports both NIF and C node should look like this

```c
#ifdef BUNDLEX_NIF
state->env=unifex_alloc_env();
#else
state->env = env;
#endif
state->reply_to = *unifex_self(env, &state->reply_to);
```
instead of this:

```c
state->env = unifex_alloc_env();
state->reply_to = *unifex_self(env, &state->reply_to);
```

Reply:
A possible solution would be to make `unifex_alloc_env` accept the original env. Then we could either copy the env or just return it (and make unifex_free_env a noop) in the c node implementation. The NIF implementation could ignore the original env.